### ISSUE TITLE
pytorch1.3.0+py36+3090 cuda11.2下面parallel_apply报错

### ISSUE THREAD

**Issue by yanhan19940405** [Link-Li/Balanced-DataParallel#9]
你好，我在进行单机4卡训练任务时，使用了你V2版本的分布式代码。结果发现程序运行到‘’return parallel_apply(replicas, inputs, kwargs, device_ids[:len(inputs)])‘’时报错。

![image](https://user-images.githubusercontent.com/11435099/108612238-8d66cf00-7421-11eb-93da-06568b5f9f56.png)
紧跟着cuda又报错了，
![image](https://user-images.githubusercontent.com/11435099/108612252-a0799f00-7421-11eb-9a0b-9bf2c8767b9b.png)
请问第一个错误是什么问题呢？

**Comment by Link-Li** [comment]
代码贴出来看一下

---原始邮件---
发件人: "Forever"<notifications@github.com&gt;
发送时间: 2021年2月21日(周日) 上午8:49
收件人: "Link-Li/Balanced-DataParallel"<Balanced-DataParallel@noreply.github.com&gt;;
抄送: "Subscribed"<subscribed@noreply.github.com&gt;;
主题: [Link-Li/Balanced-DataParallel] pytorch1.3.0+py36+3090 cuda11.2下面parallel_apply报错 (#9)




 
你好，我在进行单机4卡训练任务时，使用了你V2版本的分布式代码。结果发现程序运行到‘’return parallel_apply(replicas, inputs, kwargs, device_ids[:len(inputs)])‘’时报错。
 

 紧跟着cuda又报错了，
 
 请问第一个错误是什么问题呢？

**Comment by yanhan19940405** [comment]
你好，与分布式设置相关的代码片段如下：
`batch_szie = 16
gpu0_bsz = 6
acc_grad = 2
model = SBERT_model(encoder=bert_model,embedding_size=1024)
model=BalancedDataParallel(gpu0_bsz // acc_grad, model, dim=0)`

model部分代码如下：
class SBERT_model(torch.nn.Module):
    def __init__(self,embedding_size,encoder):
        super(SBERT_model, self).__init__()
        self.encoder= encoder
        self.emb_linear = torch.nn.Linear(1024, 512)
        self.Gele_Interaction_Modes = Gele_Interaction_Modes("ATTEND")
        self.Gele_encoder2_value = Gele_encoder2(20, "CNN")
        self.Q1_avgpool=torch.nn.AvgPool1d(10332, stride=1)
        self.Q2_avgpool = torch.nn.AvgPool1d(10332, stride=1)
        self.w=torch.nn.Parameter(torch.randn(3*128,2))

    def forward(self,Q1,Q2,attention_mask=None):
        
        mask_q1 = (Q1 != 0).float()
        mask_q2 = (Q2 != 0).float()
        #with torch.no_grad():
        Q1_encoder=self.encoder(Q1.long(),attention_mask=mask_q1)[0].float()
        Q2_encoder=self.encoder(Q2.long(),attention_mask=mask_q2)[0].float()

**Comment by yanhan19940405** [comment]
其他分布式相关的就没有了，使用方式是直接

`from balanced import BalancedDataParallel`

**Comment by yanhan19940405** [comment]
其他辅助信息还有，就是2080ti上能正常运行此代码，但迁移到3090上就不行了。。2080ti cuda驱动是10.2，440.36版本；3090上是cuda11.2，版本是460.39.
两台机器python环境都是3.6，pytorch版本都是1.3.0

**Comment by Link-Li** [comment]
因为我这里没有3090的显卡，所以没法在3090上面调试，如果你暂时可以用20系显卡，那先用20系显卡吧。这可能是cuda版本变化引起的问题。
